src/lib.rs

```diff
@@ -303,4 +277,53 @@ fn restore_extended_key(
     })
 }
 
+struct TxBuilder {
+    recipients: Vec<(String, u64)>,
+    fee_rate: Option<f32>,
+}
+
+impl TxBuilder {
+    fn new() -> Self {
+        TxBuilder {
+            recipients: Vec::new(),
+            fee_rate: None,
+        }
+    }
+
+    fn add_recipient(&self, recipient: String, amount: u64) -> Arc<Self> {
+        let mut recipients = self.recipients.to_vec();
+        recipients.append(&mut vec![(recipient, amount)]);
+        Arc::new(TxBuilder {
+            recipients,
+            fee_rate: self.fee_rate,
+        })
+    }
+
+    fn fee_rate(&self, sat_per_vb: f32) -> Arc<Self> {
+        Arc::new(TxBuilder {
+            recipients: self.recipients.to_vec(),
+            fee_rate: Some(sat_per_vb),
+        })
+    }
+
+    fn build(&self, wallet: &Wallet) -> Result<Arc<PartiallySignedBitcoinTransaction>, Error> {
+        let wallet = wallet.get_wallet();
+        let mut tx_builder = wallet.build_tx();
+        for (address, amount) in &self.recipients {
+            let address = Address::from_str(address).expect("recipient address parsing failed");
+            tx_builder.add_recipient(address.script_pubkey(), *amount);
+        }
+        if let Some(sat_per_vb) = self.fee_rate {
+            tx_builder.fee_rate(FeeRate::from_sat_per_vb(sat_per_vb));
+        }
+        tx_builder
+            .finish()
+            .map(|(psbt, _)| PartiallySignedBitcoinTransaction {
+                internal: Mutex::new(psbt),
+            })
+            .map(Arc::new)
+            .map_err(|_| BdkError::Generic("Failed to build transaction".to_string()))
```

Comment:
I think all we need to do is add the new possible BdkError enums to the `bdk.udl` file and then they won't have to be mapped to the `BdkError::Generic`.

Reply:
I think this may have been an oversight. If the error that `tx_builder.finish()` exposes is only the `BdkError` then we don't need this transform, as the `BdkError` enum is already fully described in our `bdk.udl` file. I can check again if there are other cases and if the `map_err` can be safely ignored.